### ISSUE TITLE
CommandLineArgumentProviders included as pluginOptions

### ISSUE THREAD

**Issue by cdsap** [google/ksp#1359]
Since version 1.8.10-1.0.9 we are observing cache misses when testing cache-relocatable builds: 

<img width="739" alt="229945844-a0d699db-d314-4f85-b330-c40c43ba87ce" src="https://user-images.githubusercontent.com/475733/230480481-e7c5f210-9001-4321-9c1d-f9bb70b9f123.png">

I can reproduce it with the  project `nowinandroid` where the Room processor uses Ksp, I tested in two different scenarios:
* Using the `KspWorkaround` from the Android Gradle Cache Fix Plugin
* Using the approach provided by the [Android doc](https://developer.android.com/training/data-storage/room/migrating-db-versions#export-schemas)

In both use cases, the difference lies in the apoption `com.google.devtools.ksp.symbol-processing.apoption=` 

After reviewing `KspGradleSubplugin`, looks like the issue is caused when the plugin is adding plugin arguments for the internal options defined in the KspTask:
```kotlin
kspTask.pluginOptions.add(
         project.provider {
              CompilerPluginConfig().apply {
                   (kspTask as KspTask).options.get().forEach {
                       addPluginArgument(KSP_PLUGIN_ID, it)
                   }
              }
         }
) 
```
one of these options could be a CommandLineArgumentProvider like:
``` kotlin
class RoomSchemaArgProvider(
      @get:InputDirectory
      @get:PathSensitive(PathSensitivity.RELATIVE)
      val schemaDir: File,
 ) : CommandLineArgumentProvider {
      override fun asArguments() = listOf("room.schemaLocation=${schemaDir.path}")
 }
```

I tested Ksp locally updating the type of option with `InternalSubpluginOption` for the CommandLineArguments like:
```kotlin
commandLineArgumentProviders.get().forEach {
                it.asArguments().forEach { argument ->
                    if (!argument.matches(Regex("\\S+=\\S+"))) {
                        throw IllegalArgumentException("KSP apoption does not match \\S+=\\S+: $argument")
                    }
                    options += InternalSubpluginOption("apoption", argument)
                }
            }
```
and is working. 

I was going to prepare a PR but I wanted to confirm that this approach is correct. 
Thanks

**Comment by neetopia** [comment]
closed as #1368 is merged

**Comment by bachiri** [comment]
Thanks @neetopia for merging this PR. 
In which ksp version this fix will be included? 
I just checked [1.9.0-Beta-1.0.11](https://github.com/google/ksp/releases/tag/1.9.0-Beta-1.0.11), [ 1.8.22-1.0.11](https://github.com/google/ksp/releases/tag/1.8.22-1.0.11) and didn't find it included.
